src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -2261,7 +2274,7 @@ class SharedDictionaryPrinter : StackObj {
 
   void do_value(const RunTimeSharedClassInfo* record) {
     ResourceMark rm;
-    _st->print_cr("%4d: %s %s", (_index++), record->_klass->external_name(),
+    _st->print_cr("%4d: %s %s", ++_index, record->_klass->external_name(),
```

Comment:
Not clear why the _index handling is changing.

Reply:
This is found the printout of index is not consecutive between the two dictionaries. The previous index is printout, then increased then the next dictionary take the index and increased again. I will check again.